linode/networkreservedips/framework_resource.go

```diff
@@ -0,0 +1,181 @@
+package networkreservedips
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_reserved_ip",
+				IDType: types.StringType,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Debug(ctx, "Starting Create for linode_reserved_ip")
+	var plan ReservedIPModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Error(ctx, "Error getting plan", map[string]interface{}{
+			"error": resp.Diagnostics.Errors(),
+		})
+		return
+	}
+
+	ctx = populateLogAttributes(ctx, &plan)
+
+	client := r.Meta.Client
+	reserveIP, err := client.ReserveIPAddress(ctx, linodego.ReserveIPOptions{
+		Region: plan.Region.ValueString(),
+	})
+	if err != nil {
+		tflog.Error(ctx, "Failed to reserve IP address", map[string]interface{}{
+			"error": err.Error(),
+		})
+		resp.Diagnostics.AddError(
+			"Failed to reserve IP address",
+			err.Error(),
+		)
+		return
+	}
+
+	if reserveIP == nil {
+		tflog.Error(ctx, "Received nil pointer for reserved IP")
+		resp.Diagnostics.AddError("nil Pointer", "received nil pointer of the reserved ip")
+		return
+	}
+
+	tflog.Debug(ctx, "Successfully reserved IP address", map[string]interface{}{
+		"address": reserveIP.Address,
+		"region":  reserveIP.Region,
+	})
```

Comment:
Not sure if we want to print this info into debug. It may leave some sensitive info in the log. We can instead adding a debug before the `client.ReserveIPAddress(...)` just to indicate there is an API call.